townhall/myapi/views.py

```diff
@@ -223,4 +225,60 @@ def handle_organization_update(self, request):
             serializer.save()
             return Response({"message": "Organization updated successfully"}, status=status.HTTP_200_OK)
         else:
-            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
\ No newline at end of file
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+        
+class TaskViewSet(viewsets.ViewSet):
+
+    @action(detail=False, methods=['get'])
+    def get_all_tasks(self, request):
+
+        tasks = TaskServices.get_all_tasks()
+        serializer = TaskSerializer(tasks, many=True)
+        return Response(serializer.data)
+
+    @action(detail=True, methods=['get'])
+    def get_task(self, request, pk=None):
+
+        task = TaskServices.get_task_by_id(pk)
+        if task:
+            return Response(TaskSerializer(task).data)
+        return Response({'error': 'Task not found'}, status=status.HTTP_404_NOT_FOUND)
+
+    @action(detail=False, methods=['post'])
+    def create_task(self, request, *args, **kwargs):
+      task_data = CreateTaskData(
+          name=request.data.get('name'),
+          description=request.data.get('description'),
```

Comment:
some of these may benefit from default fall back on get, so like `request.data.get('description', None)`

Reply:
Done